crates/shared/src/sources/balancer_v2/pool_cache.rs

```diff
@@ -170,10 +190,49 @@ fn is_contract_error(err: &anyhow::Error) -> bool {
     )
 }
 
+/// An internal utility method for sharing the success value for an
+/// `anyhow::Result`.
+///
+/// Typically, this is pretty trivial using `FutureExt::shared`. However, since
+/// `anyhow::Error: !Clone` we need to use a different approach.
+///
+/// # Panics
+///
+/// Polling the future with the shared success value will panic if the result
```

Comment:
This is beyond my level of rust knowledge, but is there not something we can do in terms of lifetimes to indicate that the second future has to outlive the first one?

Reply:
Maybe... I couldn't think of anything cleaver though.